### ISSUE TITLE
Github project URL is not marked as "verified" despite self-link

### ISSUE THREAD

**Issue by MartinThoma** [pypi/warehouse#16836]
**Describe the bug**

According to https://docs.pypi.org/project_metadata/ it should be possible to link to https://pypi.org/project/pypdf/ from https://github.com/py-pdf/pypdf

You can see the link from Github to the PyPI project page over the "python version" tag at the very top of the github page:

![image](https://github.com/user-attachments/assets/e71f6e12-2d39-4091-9f79-0bc4f20a3bb3)

we did exactly that, but still the URL is not marked as "verified":

![image](https://github.com/user-attachments/assets/bae9ec7a-f2b4-4c36-a594-72aa0f99fd61)

It was like this when the package was uploaded

**Expected behavior**

The source URL https://github.com/py-pdf/pypdf should be marked as verified on https://pypi.org/project/pypdf/

**To Reproduce**

https://pypi.org/project/pypdf/

**Comment by MartinThoma** [comment]
Hm ... can it be the missing trailing slash in the "Source" URL?

**Comment by woodruffw** [comment]
> According to [docs.pypi.org/project_metadata](https://docs.pypi.org/project_metadata/) it should be possible to link to [pypi.org/project/pypdf](https://pypi.org/project/pypdf/) from [py-pdf/pypdf](https://github.com/py-pdf/pypdf)

Yep -- the docs [specifically say](https://docs.pypi.org/project_metadata/#via-trusted-publishing) that you need to use a [Trusted Publisher](https://docs.pypi.org/trusted-publishers/adding-a-publisher/) to link a GitHub (or GitLab, etc.) repo to a PyPI project. Just adding a badge to the repo isn't sufficient, since anybody can add a badge claiming to be a particular project.

Once you register a Trusted Publisher and publish using it instead of a manually-configured API token, your subsequent releases will have those GitHub URLs marked as verified.

**Comment by woodruffw** [comment]
(For reference, you're currently using a manually configured API token here: https://github.com/py-pdf/pypdf/blob/fcb103a7d995406c993b7ba6b3325c94dab7ad0b/.github/workflows/publish-to-pypi.yaml#L27-L32)

**Comment by stefan6419846** [comment]
Current *pypdf* maintainer here: This was a misunderstanding and fixed a while ago on our side. Thus this issue can be closed.
